okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -517,14 +544,20 @@ void stopIfNecessary() {
     synchronized (lock) {
       if (goAway && streams.size() == 0) {
         if (!stopped) {
-          stopped = true;
-          // We will close the underlying socket in the writing thread to break out the reader
-          // thread, which will close the frameReader and notify the listener.
-          frameWriter.close();
-
-          if (ping != null) {
-            ping.failed(getPingFailure());
-            ping = null;
+          try {
+            stopped = true;
+            // We will close the underlying socket in the writing thread to break out the reader
+            // thread, which will close the frameReader and notify the listener.
+            frameWriter.close();
+
+            if (ping != null) {
+              ping.failed(getPingFailure());
+              ping = null;
+            }
+          } finally {
+            if (usingSharedExecutor) {
+              SharedResourceHolder.release(SHARED_EXECUTOR, (ExecutorService) executor);
```

Comment:
Does serializingExecutor need to be released too?  It still has a reference to it right?